### PR TITLE
Add UMAD-09 UMA Invoice

### PR DESCRIPTION
**Pull request by zhenlu**

### CODE REVIEW

**Comment by zhenlu** [comment]
* **#28** <a href="https://app.graphite.dev/github/pr/uma-universal-money-address/protocol/28?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @zhenlu and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: umad-09-invoice.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# UMAD-09 Invoice
+
+UMA invoice is a data structure that describes payments that can be coordinated using UMA and
```
Comment (jklein24) [inline umad-09-invoice.md:3]: Same note on context here that I left in the docs PR.

File: umad-09-invoice.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+```
+
+## Proof of Payment
+
+When creating the lightning invoice from the UMA invoice, the invoice UUID should be included in the
+metadata field. It should be one extra item for the metadata field. See the LUD-18 for details.
+
+```text
+["text/plain", invoiceUUID]
```
Comment (jklein24) [inline umad-09-invoice.md:87]: I'm just noticing this, but if we use text/plain, how will we differentiate from the short description? The LUD-06 [spec](https://github.com/lnurl/luds/blob/luds/06.md#wallet-to-service-interaction-flow) indicates that there must be one text/plain entry for the description. Maybe this should be text/uma-invoice  or something?

Comment (zhenlu) [reply umad-09-invoice.md:87]: Hmm I think I might have a wrong understanding that this is only for the type. But I think you are correct. I'll change it to text/uma-invoice

File: umad-09-invoice.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+provider, etc.
+
+## Invoice Format
+
+The full structure of the invoice is:
+
+```raw
+{
+    ReceiverUma: string, 
+    // Receiving UMA address
```
Comment (jklein24) [inline umad-09-invoice.md:19]: nit: All of the other code examples in the spec put doc comments above fields like JS-style fields.

File: umad-09-invoice.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    Signature: bytes
+    // The signature of the UMA invoice
+}
+
+```
+
+## Proof of Payment
+
+When creating the lightning invoice from the UMA invoice, the invoice UUID should be included in the
+metadata field. It should be one extra item for the metadata field. See the LUD-18 for details.
```
Comment (jklein24) [inline umad-09-invoice.md:89]: nit: LUD-18 is payerdata, but metadata going into the invoice is actually just from LUD-06

**Review by jklein24 — APPROVED** [review]

**Pull request merged**